packages/synapse-react-client/src/components/widgets/facet-nav/FacetPlotLegendList.tsx

```diff
@@ -23,13 +19,19 @@ export function FacetPlotLegendList(props: FacetPlotLegendProps) {
       className={`FacetNavPanel__body__legend${isExpanded ? '--expanded' : ''}`}
     >
       {labels.slice(0, numLegendItems).map((facetValue, index) => {
-        const labelDisplay = truncate(facetValue.label, MAX_LEGEND_LENGTH)
         return (
           <ElementWithTooltip
             tooltipText={facetValue.label}
             key={facetValue.label}
           >
-            <Box sx={{ display: 'flex', alignItems: 'center', gap: '10px' }}>
+            <Box
+              sx={{
+                display: 'grid',
+                gridTemplateColumns: '20px auto 30px',
```

Comment:
Change into a Grid layout, with the facet value label taking up the max space of an auto column, then being truncated via css.  This makes the truncation dynamic based on the available width.